cherrypy/process/servers.py

```diff
@@ -121,6 +121,14 @@ def index(self):
 import sys
 import time
 import warnings
+import contextlib
+
+import portend
+
+
+class Timeouts:
```

Comment:
And [in this question](https://www.youtube.com/watch?v=o9pEzgHorH0&feature=youtu.be&t=26m34s), Deiderich specifically endorses this approach as "useful."